dali/pipeline/operators/reader/nvdecoder/imgproc.cu

```diff
@@ -0,0 +1,140 @@
+#include "dali/pipeline/operators/reader/nvdecoder/imgproc.h"
+
+#include <cuda_fp16.h>
+
+namespace dali {
+
+namespace {
+
+// using math from https://msdn.microsoft.com/en-us/library/windows/desktop/dd206750(v=vs.85).aspx
+
+template<typename T>
+struct yuv {
+    T y, u, v;
+};
+
+__constant__ float yuv2rgb_mat[9] = {
+    1.164383f,  0.0f,       1.596027f,
+    1.164383f, -0.391762f, -0.812968f,
+    1.164383f,  2.017232f,  0.0f
+};
+
+__device__ float clip(float x, float max) {
+    return fmin(fmax(x, 0.0f), max);
+}
+
+template<typename T>
+__device__ T convert(const float x) {
+    return static_cast<T>(x);
+}
+
+template<>
+__device__ half convert<half>(const float x) {
+    return __float2half(x);
+}
+
+template<>
+__device__ uint8_t convert<uint8_t>(const float x) {
+    return static_cast<uint8_t>(roundf(x));
+}
+
+template<typename YUV_T, typename RGB_T>
+__device__ void yuv2rgb(const yuv<YUV_T>& yuv, RGB_T* rgb,
+                        size_t stride, bool normalized) {
+    auto mult = normalized ? 1.0f : 255.0f;
```

Comment:
More efficient way: make `normalized` a template argument and use appropriate pre-multiplied conversion matrix.